src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs

```diff
@@ -280,6 +281,13 @@ protected virtual void OnConsumedBytes(int count)
                 }
             }
 
+            // If we got here, request contains no Content-Length or Transfer-Encoding header.
+            // Reject with 411 Length Required.
+            if (HttpMethods.IsPost(context.Method) || HttpMethods.IsPut(context.Method))
```

Comment:
Nope. IIS and nginx seem to require a length only for POST and PUT, so I'm doing the same here.